### PR TITLE
e2e testing improvements

### PR DESCRIPTION
**Pull request by wilwade**
# Goal
The goal of this PR is to improve the e2e tests.

NOTE: Parallel testing locally is still flaky. Another PR coming, but wanted to keep these separate for readability

Part of #1731 

# Discussion
- Big Change: `signAndSend` now returns a correctly typed object!
    - This leads to a large number of basically the same change all over.
    - the default event (previously the `response[0]` field) now is typed, so no need to use `is` everywhere
    - Using `?` (`result?.data.msaId` for example) allows for a nice handling of the default event data
    - Using `!` (`result!.data.msaId` for example) allows for a nice throwing of the default event data when we expect it, but don't have it. (fine for tests)
- A few small cleanups like removing unused imports
- Additional helpful comments
- Fixing an issue with handles test not being able to run more than 90 times in a row

# Testing
- `make e2e-tests`
- Reading through the comments separately might be easier

### CODE REVIEW

File: e2e/handles/handles.test.ts [hunk 1]
```diff
@@ -61,20 +59,18 @@ describe("🤝 Handles", () => {
...
+            assert.notEqual(handle, "", "retireHandle should return the correct handle");
         });
     });
 
     describe("@Alt Path: Claim Handle with possible presumptive suffix/RPC test", () => {
         /// Check chain to getNextSuffixesForHandle
 
         it("should be able to claim a handle and check suffix (=suffix_assumed if available on chain)", async function () {
-            const handle = "test1";
+            const handle = "test-" + Math.random().toFixed(10).toString().replaceAll("0.", "");
```
Comment (wilwade) [inline e2e/handles/handles.test.ts:73]: While not completely foolproof, this enables our tests to run more than 90 times on Testnet

File: e2e/messages/addIPFSMessage.test.ts [hunk 2]
```diff
@@ -20,7 +19,6 @@ describe("Add Offchain Message", function () {
     let keys: KeyringPair;
     let schemaId: u16;
     let dummySchemaId: u16;
-    let messageBlockNumber: u32;
```
Comment (wilwade) [inline e2e/messages/addIPFSMessage.test.ts:23]: Unused.

File: e2e/miscellaneous/utilityBatch.test.ts [hunk 3]
```diff
@@ -12,8 +12,9 @@ describe("Utility Batch Filtering", function () {
     const fundingSource = getFundingSource("misc-util-batch");
 
     before(async function () {
-        sender = await createAndFundKeypair(fundingSource, 50_000_000n);
-        recipient = await createAndFundKeypair(fundingSource, 50_000_000n);
+        let nonce = await getNonce(fundingSource);
+        sender = await createAndFundKeypair(fundingSource, 50_000_000n, "utility-sender", nonce++);
+        recipient = await createAndFundKeypair(fundingSource, 50_000_000n, "utility-recipient", nonce++);
```
Comment (wilwade) [inline e2e/miscellaneous/utilityBatch.test.ts:17]: Parallelization bit

File: e2e/scaffolding/helpers.ts [hunk 4]
```diff
@@ -207,34 +207,26 @@ export function log(...args: any[]) {
...
-  let createProviderOp = ExtrinsicHelper.createProvider(providerKeys, "PrivateProvider");
-  let [providerEvent] = await createProviderOp.fundAndSend(source);
-  if (providerEvent && ExtrinsicHelper.api.events.msa.ProviderCreated.is(providerEvent)) {
-    providerId = providerEvent.data.providerId;
-  }
+  const providerKeys = await createAndFundKeypair(source);
+  await ExtrinsicHelper.createMsa(providerKeys).fundAndSend(source);
+  const createProviderOp = ExtrinsicHelper.createProvider(providerKeys, "PrivateProvider");
+  const { target: providerEvent } = await createProviderOp.fundAndSend(source);
+  const providerId = providerEvent?.data.providerId || new u64(ExtrinsicHelper.api.registry, 0);
```
Comment (wilwade) [inline e2e/scaffolding/helpers.ts:214]: Clean up due to being able to follow this pattern now.

File: e2e/capacity/staking.test.ts [hunk 5]
```diff
@@ -54,16 +54,10 @@ describe("Capacity Staking Tests", function () {
...
             assert.notEqual(unStakeEvent, undefined, "should return an UnStaked event");
+            assert.equal(unStakeEvent?.data.capacity, capacityMin, "should return an UnStaked event with 1 CENT reduced capacity");
 
-            if (unStakeEvent && ExtrinsicHelper.api.events.capacity.UnStaked.is(unStakeEvent)) {
-                let unstakedCapacity = unStakeEvent.data.capacity;
-                assert.equal(unstakedCapacity, capacityMin, "should return an UnStaked event with 1 CENT reduced capacity");
-            }
-            else {
-                assert.fail("should return an capacity.UnStaked.is(unStakeEvent) event");
-            }
```
Comment (wilwade) [inline e2e/capacity/staking.test.ts:66]: Good example of how the major change cleans up code.

File: e2e/scaffolding/extrinsicHelpers.ts [hunk 6]
```diff
@@ -93,14 +92,16 @@ function eventKey(event: Event): string {
  */
 
 type ParsedEvent<C extends Codec[] = Codec[], N = unknown> = IEvent<C, N>;
-export type ParsedEventResult<C extends Codec[] = Codec[], N = unknown> = [ParsedEvent<C, N> | undefined, EventMap];
+export type ParsedEventResult<C extends Codec[] = Codec[], N = unknown> = {
+    target?: ParsedEvent<C, N>;
+    eventMap: EventMap;
+};
```
Comment (wilwade) [inline e2e/scaffolding/extrinsicHelpers.ts:98]: Here is the type change!

File: e2e/scaffolding/extrinsicHelpers.ts [hunk 7]
```diff
@@ -151,7 +152,7 @@ export class Extrinsic<T extends ISubmittableResult = ISubmittableResult, C exte
         }
     }
 
-    public async fundAndSend(source: KeyringPair): Promise<ParsedEventResult> {
+    public async fundAndSend(source: KeyringPair) {
```
Comment (wilwade) [inline e2e/scaffolding/extrinsicHelpers.ts:155]: **This function is where the large set of changes is at!**

File: e2e/scaffolding/extrinsicHelpers.ts [hunk 8]
```diff
@@ -166,33 +167,33 @@ export class Extrinsic<T extends ISubmittableResult = ISubmittableResult, C exte
...
+                if (target && targetEvent.is(target)) {
+                    return { target, eventMap };
                 }
-                result[1] = em;
-                return result;
+
+                return { eventMap };
+            }),
+            tap(({ eventMap }) => {
+                Object.entries(eventMap).map(([k, v]) => log(k, v.toHuman()));
```
Comment (wilwade) [inline e2e/scaffolding/extrinsicHelpers.ts:193]: This majorly cleans up the console log when in verbose mode.

**Review by enddynayn — APPROVED** [review]
Really nice improvements! ✨

**Review by shannonwells — APPROVED** [review]
Nice work.  I jsut had the one non-blocking comment

File: e2e/handles/handles.test.ts [hunk 9]
```diff
@@ -36,12 +36,10 @@ describe("🤝 Handles", () => {
...
             const claimHandle = ExtrinsicHelper.claimHandle(msaOwnerKeys, claimHandlePayload);
-            const [event] = await claimHandle.fundAndSend(fundingSource);
+            const { target: event } = await claimHandle.fundAndSend(fundingSource);
             assert.notEqual(event, undefined, "claimHandle should return an event");
-            if (event && claimHandle.api.events.handles.HandleClaimed.is(event)) {
-                let handle = event.data.handle.toString();
-                assert.notEqual(handle, "", "claimHandle should emit a handle");
-            }
+            handle = event!.data.handle.toString();
+            assert.notEqual(handle, "", "claimHandle should emit a handle");
```
[the remark this answers is not in the document]
Comment (wilwade) [reply e2e/handles/handles.test.ts:42]: There were a lot I wanted to golf. I'll snag this update in my part 2 unless there end up being other changes needed to this one.

**Review by mattheworris — APPROVED** [review]
Nice clean up.

150 tests passing `when parallel: false`

**Pull request merged**